shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionActionDispatcher.kt

```diff
@@ -267,4 +279,34 @@ class StepCompletionActionDispatcher(
             .getState(forceUpdate = true)
             .map { it.gamification.hypercoinsBalance }
             .getOrNull()
+
+    private suspend fun handleFetchNextInterviewStep(
+        onNewMessage: (Message) -> Unit
+    ) {
+        sentryInteractor.withTransaction(
+            HyperskillSentryTransactionBuilder.buildStepCompletionNextInterviewStepLoading(),
+            onError = {
+                InternalMessage.FetchNextInterviewStepResultError(
+                    resourceProvider.getString(
+                        SharedResources.strings.step_theory_failed_to_continue_practicing
+                    )
+                )
+            }
+        ) {
+            val nextInterviewStepId = interviewStepsStateRepository
+                .getState(forceUpdate = false)
+                .getOrThrow()
+                .shuffled()
+                .firstOrNull()
+            InternalMessage.FetchNextInterviewStepResultSuccess(
+                nextInterviewStepId?.id?.let(StepRoute::InterviewPreparation)
+            )
+        }.let(onNewMessage)
+    }
+
+    private suspend fun handleMarkInterviewStepAsSolved(action: InternalAction.MarkInterviewStepAsSolved) {
+        interviewStepsStateRepository.updateState { steps ->
+            steps.mutate { remove(StepId(action.stepId)) }
+        }
+    }
```

Comment:
I saw the same code in the `InterviewPreparationWidgetActionDispatcher`, you are trying to mutate state twice.

I suggest you to subscribe for `InterviewStepsStateRepository` changes in the `InterviewPreparationWidgetActionDispatcher` and update `InterviewPreparationWidgetFeature.State.Content.steps` directly with new state